### ISSUE TITLE
singleAttributePerLine does not work as expected

### ISSUE THREAD

**Issue by iM4x** [prettier/prettier#16847]
**Prettier 3.3.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAhAEQPIGEAqAmgAoCiABABYwC2ANgHwA6UqAEvgLIAy5u3AQQDKQgLxMQUCAFoAVgGdyNQtNx0B8oROZRy5dqQGYdevak6l8AvmwEAlIZfEgArjABm0gBzaWp-fgAkvjcpAycAJ7kMMF0pKgA9EEhYX6m5pbWAHICFs4AJnDyYABOAJYADjBl0BLkkLAIMM6RfBBZMKRQ+L66+glshsZpqABGEPkRJun5ZQBu5ADEZYWwZe5lcCXRRTDSAIbO8PJ7+3XHe6NHu9Kj5zdg1yfSYPfP+U97+W97cJ-ScB+0nc-3cQIA5v9wRI0v4LtJKP9KECyv8ykDZP9ZECANb-HG9fyJWZzaaJcaTHSJDg8BggAA0IAgVRqUHkyFA+xKJQgAHdiFyEOyUPs6Lz9hF2YzRiV9mAcXAYEJ9jQ4NwylA4Mh3KL5HBpbL5YqhBU5RrwcgYCUXPqQHqaGVLdbbXAAB4VLZlVWwUX4LZQLmbYU6uh6xnyc10OAARRcEHg2t1toUrqEkZjcYTSBDYZAAEdM3BiDyKsKQPt5NJNXBCvkGSArfsynRzbgIDQaPtkOW6HR6xGoOCowIYFayqM3EWturNYnQ7bqPQAOqUMrHU1gOBCIVr+ZriLdsDyKUgOY2wJQVZK0qVGACS9CGARKNz3MVHl6peyird99FLZzFqjIanqJQwMW+zgp2r62qaJSgd2i59oy74ajAS4rDAlDIAAjAALAADIyJRwAWZQkRBUFdtmSaMjA+yjBh+RYcgABMjIuHq+AMcGtEgHANCjDWtbcPsg4uJBcAAGIQCUnajua3b7G4ED1mALjwbJ2DuO4eowMgeE4axAC+xlAA)

```sh
--parser html
--print-width 140
--cursor-offset 412
```

**Input:**

```html
<!DOCTYPE html>
<HTML CLASS="no-js mY-ClAsS">
  <HEAD>
    <META CHARSET="utf-8">
    <TITLE>My tITlE</TITLE>
    <META NAME="description" content="My CoNtEnT">
  </HEAD>
  <body>
    <div #identifier test-a="test-a" test-b="test-b" test-c="test-c" test-d="test-d" test-e="test-e" test-f="test-f" test-g="test-g"
      test-h="test-h" test-i="test-i" test-j="test-j" test-k="test-k">
    </div>
  </body>
</HTML>
```

**Output:**

```html
<!doctype html>
<html class="no-js mY-ClAsS">
  <head>
    <meta charset="utf-8" />
    <title>My tITlE</title>
    <meta name="description" content="My CoNtEnT" />
  </head>
  <body>
    <div
      #identifier
      test-a="test-a"
      test-b="test-b"
      test-c="test-c"
      test-d="test-d"
      test-e="test-e"
      test-f="test-f"
      test-g="test-g"
      test-h="test-h"
      test-i="test-i"
      test-j="test-j"
      test-k="test-k"
    ></div>
  </body>
</html>

```

**Expected output:**

```html
<!DOCTYPE html>
<HTML CLASS="no-js mY-ClAsS">
  <HEAD>
    <META CHARSET="utf-8">
    <TITLE>My tITlE</TITLE>
    <META NAME="description" content="My CoNtEnT">
  </HEAD>
  <body>
    <div #identifier test-a="test-a" test-b="test-b" test-c="test-c" test-d="test-d" test-e="test-e" test-f="test-f" test-g="test-g"
      test-h="test-h" test-i="test-i" test-j="test-j" test-k="test-k">
    </div>
  </body>
</HTML>
```

**Why?**

I would expect that the "singleAttributePerLine" option prints all html attributes inline when its false and only breaks, when "printWidth" is reached. At least i would expect an option to configure this behaviour to prevent my code from looking like it currently does after formatting with prettier.
Besides that I love the work you did, but can't use Prettier to format my html, because in my angular projects my html looks very unreadable.

**Comment by parked-toes** [comment]
Any news? This prevents me to user prettier in Angular project

**Comment by murilosdv** [comment]
Any news on this one? I'm having the same issue.

**Comment by maxschroers** [comment]
We've recently started using the new Angular control flow syntax (@if(), @for(), etc.) in our project. Until now, we’ve been relying on the built-in VSCode formatter, but it doesn’t properly handle formatting for these new constructs. However, it doesn't properly support formatting for these new constructs and also lacks a CLI option for formatting the entire project automatically.

We’d love to start using Prettier instead, as it aligns more closely with the formatting behavior we need—specifically, the behavior described in this issue.

**Comment by shrpne** [comment]
I don't see `singleAttributePerLine` enabled (by default it is disabled) in the provided example

**Comment by davidgiessing** [comment]
@shrpne that is exactly the point isn't it? If it is disabled by default it should not break on each attribute

**Comment by shrpne** [comment]
@davidgiessing Ah, i see. Sorry, I was tricked by a misleading title. So the `singleAttributePerLine: false` works as expected, it's just not enforce `allAttributesInline` what topicstarter expects

**Comment by loicmncqt** [comment]
Be careful with your config location, it may be overrided by the package.json. 

Check it with :
`npx prettier --find-config-path src/app/app.component.html`

Change your "src/app/app.component.html" with the path of one of your html file.
Config option _singleAttributePerLine_ works for me with prettier 3.7.3

**Comment by dgiessing** [comment]
@loicmncqt I cannot seem to find the _allAttributesInline_ Config Option?

**Comment by loicmncqt** [comment]
You're right, it seems this config option doesn't exist 😅. I edited my previous post

**Comment by rekhubs** [comment]
Exact same issue for me! Angular html. Look forward to some updates on this...
